components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -80,94 +63,74 @@ namespace scream
 class AtmosphereInput 
 {
 public:
-  using dofs_list_type = AbstractGrid::dofs_list_type;
+  using fm_type   = FieldManager<Real>;
+  using grid_type = AbstractGrid;
+
+  using KT = KokkosTypes<DefaultDevice>;
   template<int N>
-  using view_ND_host = typename KokkosTypes<DefaultDevice>::template view_ND<Real,N>::HostMirror;
-  using view_1d_host = view_ND_host<1>;
+  using view_Nd_host = typename KT::template view_ND<Real,N>::HostMirror;
+  using view_1d_host = view_Nd_host<1>;
 
   // --- Constructor(s) & Destructor --- //
-  AtmosphereInput (const ekat::Comm& comm, const ekat::ParameterList& params,
-                   const std::shared_ptr<const FieldManager<Real>>& field_mgr,
-                   const std::shared_ptr<const GridsManager>& grid_mgr)
-    : m_params    (params)
-    , m_comm      (comm)
-    , m_field_mgr (field_mgr)
-    , m_grid_mgr  (grid_mgr)
-  {
-    // Nothing to do here
-  }
-
-  AtmosphereInput (const ekat::Comm& comm, const std:: string grid_name,
-                   const std::shared_ptr<const GridsManager>& grid_mgr)
-    : m_comm      (comm)
-    , m_grid_mgr  (grid_mgr)
-    , m_grid_name (grid_name)
-  {
-    // Nothing to do here
-  }
+  AtmosphereInput (const ekat::Comm& comm,
+                   const ekat::ParameterList& params);
+  AtmosphereInput (const ekat::Comm& comm,
```

Comment:
Do any of these constructors need explanation?

Reply:
Probably. I will add some.